File: Java/Plasma/plCreatable.java
package Plasma;

import java.io.IOException;
import Plasma.CoreLib.hsStream;

public abstract class plCreatable {
    public abstract void read(hsStream S) throws IOException;
    public abstract void write(hsStream S) throws IOException;

    public abstract int getTypeNum();
    public abstract String getTypeName();
}

File: C++/plCreatable.cpp
#include "plCreatable.h"

plCreatable::plCreatable() { }
plCreatable::~plCreatable() { }

void plCreatable::read(hsStream *S) { }
void plCreatable::write(hsStream *S) { }

const char* plCreatable::getTypeName() { return ""; }
const int plCreatable::getTypeNum() { return -1; }

